Analysis/HistProducerFile.py

```diff
@@ -0,0 +1,183 @@
+                    if treeName_Valid in file_keys:
+                        dfWrapped_Valid = DataFrameBuilder(ROOT.RDataFrame(treeName_Valid, inFile))
+                        var_list = []
+                        dfWrapped_Valid.CreateFromDelta(var_list, dfWrapped_central.colNames, dfWrapped_central.colTypes)
+                        PrepareDfWrapped(dfWrapped_Valid)
+                        all_dataframes[key_2].append(dfWrapped_Valid.df)
+
+                    treeName_nonValid = f"{treeName}_nonValid"
+                    if treeName_nonValid in file_keys:
+                        dfWrapped_nonValid = DataFrameBuilder(ROOT.RDataFrame(treeName_nonValid, inFile))
+                        PrepareDfWrapped(dfWrapped_nonValid)
+                        all_dataframes[key_2].append(dfWrapped_nonValid.df)
+                else:
+                    all_dataframes[key_2] = all_dataframes[key_central]
+
+
+    print(f"time to get all dataframes dict is {datetime.now() - startTime}")
+    all_histograms = GetHistogramDictFromDataframes(all_dataframes, hist_cfg_dict,dataset)
+    print(f"time to get all histograms dict is {datetime.now() - startTime}")
+    return all_histograms
+
+
+if __name__ == "__main__":
+    import argparse
+    import yaml
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--inFile', required=True, type=str)
+    parser.add_argument('--outDir', required=False, type=str)
+    parser.add_argument('--dataset', required=True, type=str)
+    parser.add_argument('--test', required=False, type=bool, default=False)
+    parser.add_argument('--deepTauVersion', required=False, type=str, default='v2p1')
+    parser.add_argument('--compute_unc_variations', type=bool, default=False)
+    parser.add_argument('--compute_rel_weights', type=bool, default=False)
+    parser.add_argument('--histConfig', required=True, type=str)
+    parser.add_argument('--furtherCut', required=False, type=str, default = "")
+    args = parser.parse_args()
+    startTime = datetime.now()
+    headers_dir = os.path.dirname(os.path.abspath(__file__))
+    ROOT.gROOT.ProcessLine(f".include {os.environ['ANALYSIS_PATH']}")
+    ROOT.gInterpreter.Declare(f'#include "include/HistHelper.h"')
+    ROOT.gInterpreter.Declare(f'#include "include/Utilities.h"')
+    if not os.path.isdir(args.outDir):
+        os.makedirs(args.outDir)
+
+    inFile_idx_list = args.inFile.split('/')[-1].split('.')[0].split('_')
+    inFile_idx = inFile_idx_list[1] if len(inFile_idx_list)>1 else 0
+    hist_cfg_dict = {}
+    unc_cfg_dict = {}
+    unc_cfg = os.path.join(os.environ['ANALYSIS_PATH'],"config/weight_definition.yaml")
+    with open(unc_cfg, 'r') as f:
+        unc_cfg_dict = yaml.safe_load(f)
+    with open(args.histConfig, 'r') as f:
+        hist_cfg_dict = yaml.safe_load(f)
+    vars_to_plot = list(hist_cfg_dict.keys())
+    sample_cfg_dict = {}
+    sample_cfg = os.path.join(os.environ['ANALYSIS_PATH'],"config/samples_Run2_2018.yaml")
```

Comment:
Pass `unc_cfg` and `sample_cfg` as command line arguments

Reply:
ok